### DISCUSSION TITLE
Best way to interpolate cubed sphere grid to a regular lat-lon grid?

### DISCUSSION THREAD

**Discussion by ali-ramadhan** [CliMA/Oceananigans.jl#3003] [General]
This would be pretty useful for outputting cubed sphere fields for visualization purposes.

We may want a separate method for writing raw cubed sphere field data in case some users want to close budgets or do some more careful analysis.

We might need the `VectorField` abstraction (#1585) first as velocities on different faces might have to rotated first before interpolating and outputting them.

ClimateMachine.jl might also want (or already have?) such a way of interpolating cubed sphere data to a lat-lon grid? I guess it's somewhat you might want to do for any curvilinear grid on the sphere.

cc @valeriabarra @simonbyrne

**Comment by glwagner** [comment]
Will the interpolation utilities we've built for Lagrangian particle advection work for this?

**Comment by valeriabarra** [comment]
Hi @ali-ramadhan , sorry for my late comment. I am not sure if and/or how we have a conversion to lat-long for viz purposes. Perhaps @kpamnany knows? 
I know that @simonbyrne is re-working our coordinate system entirely in the ClimateMachineCore.jl repo
